jerry-core/parser/js/js-scanner-util.c

```diff
@@ -802,6 +809,14 @@ scanner_pop_literal_pool (parser_context_t *context_p, /**< context */
         {
           type = SCANNER_STREAM_TYPE_DESTRUCTURED_ARG;
         }
+
+        if (literal_p->type & SCANNER_LITERAL_IS_VAR)
+        {
+          type = (uint8_t) (type + 1);
```

Comment:
`type++`?

Reply:
That is not working, becasue type is converted to int before ++, and int cannot be assigned to byte without casting when werror is enabled.